WordPress/gutenberg
# Hide block toolbar when spacing visualizer is showing

## What?
Hides the selected block toolbar when the padding/margin visualiser is showing.

## Why?
Part of the visualizer improvements from #44700

## How?
Adds new store action/reducer/selector to indicate when the spacing visualizer is showing, and uses the selector to show/hide the toolbar

## Testing Instructions

- Add a group block and adjust the margins and padding
- Make sure the block toolbar disappears when the visualizer appears

## Screenshots or screencast 

Before:

https://user-images.githubusercontent.com/3629020/196817894-4c39a498-4b70-4652-984d-567c23bba862.mp4

After:

https://user-images.githubusercontent.com/3629020/196817916-ea9e103e-6794-4d4c-847a-a1d048e4c970.mp4
